executor/new_executor_xapi.go

```diff
@@ -229,69 +226,79 @@ func (e *NewXSelectIndexExec) indexRowToTableRow(handle int64, indexRow []types.
 
 func (e *NewXSelectIndexExec) nextForDoubleRead() (*Row, error) {
 	if e.tasks == nil {
-		startTs := time.Now()
-		handles, err := e.fetchHandles()
+		idxResult, err := e.doIndexRequest()
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		log.Debugf("[TIME_INDEX_SCAN] time: %v handles: %d", time.Now().Sub(startTs), len(handles))
-		e.buildTableTasks(handles)
-		limitCount := int64(-1)
-		if e.indexPlan.LimitCount != nil {
-			limitCount = *e.indexPlan.LimitCount
-		}
-		concurrency := 2
-		if limitCount == -1 {
-			concurrency = len(e.tasks)
-		} else if limitCount > int64(BaseLookupTableTaskSize) {
-			concurrency = int(limitCount/int64(BaseLookupTableTaskSize) + 1)
-		}
-		log.Debugf("[TIME_INDEX_TABLE_CONCURRENT_SCAN] start %d workers", concurrency)
-		e.runTableTasks(concurrency)
+		idxResult.Fetch()
+
+		// use a background goroutine to fetch index, put the result in e.tasks.
+		// e.tasks serves as a pipeline, so fetch index and get table data would
+		// run concurrency.
+		e.tasks = make(chan *lookupTableTask, 10)
```

Comment:
in the thread pool model, there is a queue,  and a set of threads which are the workers.
mostly when we meet the word "concurrency", it means the number of workers.
while the "10" here is the capacity of queue...